vaapi/vaapiimagepool.cpp

```diff
@@ -69,6 +69,7 @@ struct VaapiImagePool::ImageRecycler
         if (!image)
             return;
         m_pool->recycleID(image->getID());
+        m_pool.reset();
```

Comment:
From my understanding, it should release the use_count automatically after void operator()(VaapiImage* image) called. But it's not meet your observation. Anyway, I will spend half day to dig more. If I did not find more useful things. I will accept your patch. Is it ok to you?